### DISCUSSION TITLE
How do I get the search params in Korge for JS?

### DISCUSSION THREAD

**Discussion by Kietyo** [korlibs/korge#2201] [Q&A] [answered]
So for example if the user visits:

http://localhost:60707/index.html?state=state1234&code=blahblah&scope=profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile

How do I get the URL search params: `?state=state1234&code=blahblah&scope=profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile`?

**Comment by Kietyo** [comment, marked as answer]
In your common source directory, you can define a function like this:

```kotlin
expect fun getUrlSearchParams(): String
```

Then you can implement target specific implementations.

jvm:

```kotlin
// XenoTactic-GameLogic/src/jvmMain/kotlin/com.xenotactic.gamelogic/utils/utils.jvm.kt
actual fun getUrlSearchParams(): String {
    return ""
}
```

js:

```kotlin
// XenoTactic-GameLogic/src/jsMain/kotlin/com.xenotactic.gamelogic/utils/utils.js.kt
import kotlinx.browser.window

actual fun getUrlSearchParams(): String {
    return window.location.search
}
```

You can use this function to get the url search params, here's an example of what it looks like in the console:

<img width="1224" alt="image" src="https://github.com/korlibs/korge/assets/5054996/8ec040b4-1212-4d3d-a9d0-f92a4baa08c5">

**Comment by soywiz** [comment]
There is also `Environment`. In the case of JS, parameters are available there:

https://github.com/korlibs/korge/blob/b1bc0ae9295aeb38b65b34b9b0176910d94fe23c/korlibs-io/src%40js/korlibs/io/lang/EnvironmentJs.kt#L10

https://github.com/korlibs/korge/blob/b1bc0ae9295aeb38b65b34b9b0176910d94fe23c/korlibs-io/src%40js/korlibs/io/runtime/browser/JsRuntimeBrowser.kt#L41-L42

You can use for example:

```
Environment["scope"]
```
